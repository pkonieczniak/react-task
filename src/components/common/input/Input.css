.control-group {
  margin: 40px 20px;
  position: relative;
}

.control-group input[type="text"] {
  -webkit-appearance: none;
  appearance: none;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  border: 5px solid #f0f0f0;
  background: #fff;
  border-radius: 0;
  color: #222;
  font-family: "Open Sans", sans-serif;
  font-size: 20px;
  font-weight: 600;
  outline: none;
  padding: 13px 12px 16px 5px;
  transition-duration: 500ms;
  transition-property: border-color, padding-top, padding-bottom;
  width: 100%;
}

.control-group input[type="text"]::placeholder {
  color: #9c9c9c;
  font-weight: 400;
}

.control-group input[type="text"]:focus {
  border-color: #2d9ae3;
  padding-top: 24px;
  padding-bottom: 5px;
}
