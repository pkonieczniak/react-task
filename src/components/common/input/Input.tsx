import React, { FC } from "react";
import "./Input.css";

interface InputProps {
  name: string;
  type: string;
  onChange(): void;
  placeholder?: string;
  value?: string;
}

export const Input: FC<InputProps> = ({
  name,
  type,
  onChange,
  placeholder,
  value,
}) => {
  return (
    <div className="control-group">
      <input
        id={name}
        name={name}
        onChange={onChange}
        placeholder={placeholder}
        type={type}
        value={value}
        data-testid="input"
      />
    </div>
  );
};
