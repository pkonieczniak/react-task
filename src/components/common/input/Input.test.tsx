import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Input } from "./Input";

test("renders Input component", () => {
  const callback = jest.fn();
  const { getByTestId } = render(
    <Input
      name="foo"
      type="text"
      placeholder="lorem ipsum"
      onChange={callback}
    />
  );

  const inputElement = getByTestId("input") as HTMLInputElement;
  fireEvent.change(inputElement, { target: { value: "foo" } });

  expect(inputElement.value).toBe("foo");
  expect(callback).toHaveBeenCalled();
  expect(inputElement).toMatchSnapshot();
});
