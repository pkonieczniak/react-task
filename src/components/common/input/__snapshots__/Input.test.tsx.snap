// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input component 1`] = `
<input
  data-testid="input"
  id="foo"
  name="foo"
  placeholder="lorem ipsum"
  type="text"
  value=""
/>
`;
